tests/testResolutionsClass.m

```diff
@@ -0,0 +1,283 @@
+classdef testResolutionsClass < matlab.unittest.TestCase   
+    
+    properties
+        parameters
+        resolutions
+    end
+
+    methods(TestMethodSetup)
+        function createResolutions(testCase)
+            testCase.parameters = {
+                    % Name               min   val   max  fit? 'Prior Type','mu','sigma'
+                    'Resolution par 1', 0.01, 0.03, 0.05, false, 'uniform', 0, Inf;
+                    'Resolution par 2', 0.1, 0.19, 1.0, true, 'gaussian', -1, 1;
+                    'Resolution par 3', 0.2, 0.17, 1.1, false, 'uniform', 0, Inf;
+                    };
+            testCase.resolutions = {
+                    'Resolution 1', 'gaussian', 'Resolution par 1', '', '', '', '';
+                    'Resolution 2', 'constant', 'Resolution par 2', '', '', '', '';
+                    'Resolution 3', 'function', 'function_name', 'Resolution par 1', 'Resolution par 2', '', '';
+                    'Resolution 4', 'data', '', '', '', '', '';
+                    };
+        end
+    end
+
+    methods (Test)
+        function testCreation(testCase)
+            % Tests resolution class can be created and the start parameters is set correctly  
+            resolution = resolutionsClass(testCase.parameters(1, :), testCase.resolutions(1, :));
+            testCase.verifyEqual(string(resolution.resolPars.paramsTable{1, :}), ...
+                                 string(testCase.parameters(1, :)), 'Start resolution parameter not set correctly');
+            testCase.verifySize(resolution.resolPars.paramsTable, [1, 8], 'Resolution Parameters has wrong dimension');
+            
+            testCase.verifyEqual(string(resolution.resolutions.typesTable{1, :}), ...
+                                 string(testCase.resolutions(1, :)), 'Start resolution parameter not set correctly');
+            testCase.verifySize(resolution.resolutions.typesTable, [1, 7], 'Resolutions has wrong dimension');
+        end
+        
+        function testShowPrior(testCase)
+            % Tests showPrior property
+            resolution = resolutionsClass(testCase.parameters(1, :), testCase.resolutions(1, :));
+            testCase.verifyFalse(resolution.showPriors);
+            resolution.showPriors = true;
+            testCase.verifyTrue(resolution.showPriors);
+            testCase.verifyTrue(resolution.resolPars.showPriors);
+            testCase.verifyError(@setShowPriors, invalidType.errorID);  % showPrior should be logical 
+            function setShowPriors
+                resolution.showPriors = 'a';
+            end
+            resolution.showPriors = true;
+        end
+        
+        function testGetResolNames(testCase)
+            % Tests getResolNames returns correctly  
+            resolution = resolutionsClass(testCase.parameters(1, :), testCase.resolutions(1, :));
```

Comment:
It would be worth defining a resolutions class in the method setup, given that this line appears in several tests.